### PR TITLE
Added if_indextoname function.

### PR DESCRIPTION
**Pull request by Frostie314159**
## Checklist:

- [x] I have read `CONTRIBUTING.md`
- [x] I have written necessary tests and rustdoc comments
- [x] A change log has been added if this PR modifies nix's API

### CODE REVIEW

**Comment by SteveLauC** [comment]
Hi, thanks for your interest in contributing to Nix!

CI failures are not related, so no worries, I will take a closer look at this PR this weekend:)

File: src/net/if_.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@ pub type IflagsType = libc::c_int;
 /// type alias for InterfaceFlags
 pub type IflagsType = libc::c_longlong;
 
-/// Resolve an interface into a interface number.
+/// Resolve an interface into an interface number.
```
Comment (SteveLauC) [inline src/net/if_.rs:17]: Thanks for this!

File: src/net/if_.rs [hunk 2]
```diff
@@ -26,6 +26,22 @@ pub fn if_nametoindex<P: ?Sized + NixPath>(name: &P) -> Result<c_uint> {
     }
 }
 
+/// Resolve an interface number into an interface.
+pub fn if_indextoname(index: c_uint) -> Result<CString> {
+    // We need to allocate this anyway, so doing it directly is faster.
+    let mut buf = vec![0u8; IF_NAMESIZE];
+
+    let return_buf = unsafe {
+        libc::if_indextoname(index, buf.as_mut_ptr() as *mut _)
```
Comment (SteveLauC) [inline src/net/if_.rs:35]: For pointer casting, [we prefer to use `.cast()` over the `as` keyword](https://github.com/nix-rust/nix/blob/master/CONVENTIONS.md#pointer-type-casting):

```suggestion
        libc::if_indextoname(index, buf.as_mut_ptr().cast())
```

File: src/net/if_.rs [hunk 3]
```diff
@@ -26,6 +26,22 @@ pub fn if_nametoindex<P: ?Sized + NixPath>(name: &P) -> Result<c_uint> {
...
+    };
+
+    if return_buf.is_null() {
+        Err(Error::last())
+    } else {
+        Ok(CStr::from_bytes_until_nul(buf.as_slice()).unwrap().to_owned())
+    }
```
Comment (SteveLauC) [inline src/net/if_.rs:42]: Nix has a nice helper function to help you handle those common syscall error cases:

```suggestion
   nix::errno::Errno::result(return_buf)?;
   Ok(CStr::from_bytes_until_nul(buf.as_slice()).unwrap().to_owned())
```

**Review by SteveLauC — CHANGES_REQUESTED** [review]
Hi, this PR looks good! I have left some comments on those minor style things, thanks again for this!

File: src/net/if_.rs (hunk 2 above)
Comment (Frostie314159) [reply src/net/if_.rs:35]: Ok, I'm going to fix this later today.

File: src/net/if_.rs (hunk 3 above)
Comment (Frostie314159) [reply src/net/if_.rs]: I had to add a `.cast()` to the `return_buf`, since `ErrnoSentinel` is only implemented for `*mut c_void`.

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Comment by SteveLauC** [comment]
CI issues fixed in #2344, looks like you need to rebase your branch:)

**Comment by Frostie314159** [comment]
I just rebase and MacOS is still failing.

**Comment by SteveLauC** [comment]
Thanks!

**Pull request merged**